src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -324,14 +328,15 @@ IRawElementProviderSimple* HwndTerminal::_GetUiaProvider() noexcept
 {
     // If TermControlUiaProvider throws during construction,
     // we don't want to try constructing an instance again and again.
-    // _uiaProviderInitialized helps us prevent this.
-    if (!_uiaProviderInitialized)
+    if (!_uiaProvider)
     {
         try
         {
             auto lock = _terminal->LockForWriting();
-            LOG_IF_FAILED(::Microsoft::WRL::MakeAndInitialize<::Microsoft::Terminal::TermControlUiaProvider>(&_uiaProvider, this->GetUiaData(), this));
-            _uiaProviderInitialized = true;
+            LOG_IF_FAILED(::Microsoft::WRL::MakeAndInitialize<HwndTerminalAutomationPeer>(&_uiaProvider, this->GetUiaData(), this));
+            _uiaEngine = std::make_unique<::Microsoft::Console::Render::UiaEngine>(_uiaProvider.Get()); // TODO CARLOS: fix circular ref

```

Comment:
circular ref -- OK? memory leak?

Reply:
I think this was left over from an earlier iteration. Should be good. We have a `unique_ptr<UiaEngine>` pointing to a `ComPtr<HwndTerminalAP>`. But that `HwndTerminalAP` doesn't point back.